### ISSUE TITLE
Feature Request: `no-dupe-keys` covers classes.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3294]
## test.js

```js
/* eslint no-dupe-keys: 2 */
({ foo(){}, foo(){} });
class A { foo(){} foo(){} }
```

## Actual

```
> eslint -v
v1.0.0
> eslint test.js --env es6 --no-eslintrc

test.js
  2:2  error  Duplicate key 'foo'  no-dupe-keys

✖ 1 problem (1 error, 0 warnings)
```

## Expected

```
test.js
  2:2  error  Duplicate key 'foo'  no-dupe-keys
  3:1  error  Duplicate key 'foo'  no-dupe-keys

✖ 2 problem (2 error, 0 warnings)
```

Note: allows same names between static and non static.

**Comment by nzakas** [comment]
I'm not sure this should work on classes because they aren't object literals. Maybe a new rule?

**Comment by michaelficarra** [comment]
Who would ever want to configure them individually? Really, nobody is going to want to disable either of these rules. They are clearly always errors. My vote is for just updating the existing rule to support classes.

**Comment by mysticatea** [comment]
I thought the name of `no-dupe-keys` focuses on `keys` (== properties), and method definitions also are `keys`. So I believe `no-dupe-keys` should cover classes.

However, I'm worried that this is a breaking change or not.

**Comment by nzakas** [comment]
We've used "keys" to mean keys in object literals throughout the project. I think expanding that definition would lead to confusion.

**Comment by mysticatea** [comment]
I got it.
Well, how about `no-dupe-class-members`?

**Comment by nzakas** [comment]
:+1:  that would also cover us in the case of properties in the future

**Comment by michaelficarra** [comment]
The spec calls them ClassElements btw.

**Comment by mysticatea** [comment]
I'm not sure whether or not "elements" is a common word in this case.
